proposals/NNNN-java-suite-cleanup.md

```diff
@@ -0,0 +1,116 @@
+
+There is also a learning curve in trying to understand the current project structure for both new developers and occasional contributors. The existing structure was only meant to be a bridge until we could make a breaking change. Now that we have more people contributing to the project, it's important to make it as easy as possible for them to understand and be able to effectively contribute.
+
+## Proposed solution
+
+### Remove requirement for baseAndroid folder
+The `baseAndroid` folder contains symbolic links to certain source files and folders. This was done in order to reuse as much code as possible without causing a breaking change in the Android sub-project of the Java Suite. It works well on Unix-based systems, however, Windows doesn't like the symbolic links and requires an extra script to create them on the machine during compilation. It is a less than ideal process.
+
+If we create a breaking change, we can move the Android sub-project to all the same layers as the Java SE and EE projects. This will allow the Android sub-project to use the same `base` source set folder and remove the symbolic link source set folder that is currently used.
+
+### Consistent Formatting
+Over the lifetime of the library, some coding standards have been missed or changed. The library should have a formatter run on it to provide consistent formatting to all files. This will likely touch most files and could be seen as a very large change, but while the source files will change, the code itself will remain the same. These changes will also be added to a coding style guide in the Java Suite repository.
+
+1. Copyright should be at the top of all files in a comment block before all code.
+2. Spaces not tabs. All tabbed indents will be changed to four spaces.
+3. All public methods should have JavaDoc 
+4. Correct camel-case for code objects. 
+    1. All methods and local variables should be camel-case starting with a lower-case letter (e.g. sdlManager).
+    2. All class names should be camel-case starting with an upper-case letter (e.g. SdlSession, ScreenManager).
+    3. All class packages should be lower case with no special characters (e.g. com.smartdevicelink.managers).
+5. Proper spacing between arguments and keywords. For example
+
+Wrong:
+```java
+if(x==y)
+{
+    z=1;
+}
+else{
+    z=0;
+}
+``` 
+
+Right:
+```java
+if (x == y) {
+    z = 1;
+} else {
+    z = 0;
+}
+``` 
+
+### Remove Deprecated Code
+Most or all deprecated code will be removed at the project maintainers discretion. This will include the all the old layers of `SdlConnection`, `SdlSession` (Android version), and `WiProProtocol`. Since the acceptance and integration of proposal [SDL 0194 - Android Transport Layer Overhaul](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0194-android-transport-overhaul.md), these layers are no longer needed and only bloat the library while preventing the use of the `base` source set folder.
```

Comment:
```suggestion
Most or all deprecated code will be removed at the Project Maintainer's discretion. This will include all the old layers of `SdlConnection`, `SdlSession` (Android version), and `WiProProtocol`. Since the acceptance and integration of proposal [SDL 0194 - Android Transport Layer Overhaul](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0194-android-transport-overhaul.md), these layers are no longer needed and only bloat the library while preventing the use of the `base` source set folder.
```